### PR TITLE
Fix config diff test

### PR DESCRIPTION
**Pull request by sknot-rh**
### Type of change
- Bugfix

### Description
Use reflection to call package-protected ctor to fix tests.

### Checklist

- [ ] Write tests
- [x] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java [hunk 1]
```diff
@@ -254,10 +271,10 @@ public void testChangedListenerSecurityProtocolMapFromNonDefault() {
 
     @Test
     public void testChangedMoreProperties() {
-        ArrayList<ConfigEntry> ces = new ArrayList<>();
+        ArrayList<ConfigEntry> ces = new ArrayList<>(3);
         ces.add(new ConfigEntry("inter.broker.listener.name", "david"));
         ces.add(new ConfigEntry("group.min.session.timeout.ms", "42"));
-        ces.add(new ConfigEntry("host.name", "honza"));
+        ces.add(new ConfigEntry("zookeeper.sync.time.ms", "8000"));
```
Comment (tombentley) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java:277]: Can we add comments for why we chose these three configs specifically?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java [hunk 2]
```diff
@@ -37,6 +39,17 @@
...
+    // use reflection to instantiate ConfigEntry
+    private Constructor constructor;
+    {
+        try {
+            constructor = ConfigEntry.class.getDeclaredConstructor(String.class, String.class, ConfigEntry.ConfigSource.class, boolean.class, boolean.class, List.class, ConfigEntry.ConfigType.class, String.class);
+            constructor.setAccessible(true);
+        } catch (NoSuchMethodException e) {
+            fail();
+        }
+    }
```
Comment (tombentley) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java:51]: There doesn't seem to be any benefit from having this as a field. I would just factor out a `createConfigEntry` method which did the `getDeclaredConstructor` and `newInstance` in one place.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java (hunk 1 above)
Comment (sknot-rh) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java:277]: There is no specific reason. We are doing just 3 random changes. Property `host.name` was removed from kafka 3.0.0 config model, so we just switched to another property.

**Review by scholzj — COMMENTED** [review]
Do you need to uncomment the `@Disabled` to actually run these tests in the CI?

**Review by tombentley — APPROVED** [review]

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java [hunk 3]
```diff
@@ -29,14 +30,31 @@
...
-@Disabled
 public class KafkaBrokerConfigurationDiffTest {
 
     private static final KafkaVersion.Lookup VERSIONS = KafkaVersionTestUtils.getKafkaVersionLookup();
     private static final String KAFKA_VERSION = "3.0.0";
     KafkaVersion kafkaVersion = VERSIONS.version(KAFKA_VERSION);
     private int brokerId = 0;
 
+
+
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java:41]: Sorry, one more nit ... do we need the tripple empty line?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java (hunk 1 above)
Comment (scholzj) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java:277]: Might be still worth a comment that they are randomly chosen without any deeper meaning ... even that is an information which might be useful.

**Review by scholzj — APPROVED** [review]

**Pull request merged**